### PR TITLE
Api fixes

### PR DESCRIPTION
**Pull request by rabstejnek**
Updates to API after testing openapi schema.

### CODE REVIEW

File: hawc/apps/assessment/serializers.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ class AssessmentRootedSerializer(serializers.ModelSerializer):
     NO_PARENT = -1
 
     def get_parent(self, assessment_id, validated_data, canSelectRoot):
-        parent_id = validated_data.pop("parent")
+        parent_id = validated_data.pop("parent", self.NO_PARENT)
```
Comment (rabstejnek) [inline hawc/apps/assessment/serializers.py:67]: There was an error if no `parent` was given; if you don't think this is a suitable default, we can try to make it required instead.

File: hawc/apps/lit/serializers.py [hunk 2]
```diff
@@ -224,7 +224,7 @@ def to_representation(self, instance):
         assessment_id = self.context["assessment"].id
         root = models.ReferenceFilterTag.get_assessment_root(assessment_id)
         tree = root.dump_bulk(root, keep_ids=True)
-        return {"tree": tree[0]["children"]}
+        return {"tree": tree[0].get("children")}
```
Comment (rabstejnek) [inline hawc/apps/lit/serializers.py:227]: If just the root tag w/o children then `children` wouldn't be set; this circumvents the KeyError that was thrown.

File: hawc/apps/lit/serializers.py [hunk 3]
```diff
@@ -372,6 +372,9 @@ def update(self, instance, validated_data):
         # updates the reference tags
         if "tags" in validated_data:
             instance.tags.set(validated_data.pop("tags"))
+        # updates the searches
```
Comment (rabstejnek) [inline hawc/apps/lit/serializers.py:375]: `searches` is required for PUT requests, so this allows them to be set.

File: hawc/apps/mgmt/serializers.py [hunk 4]
```diff
@@ -3,13 +3,17 @@
...
 from ..myuser.serializers import HAWCUserSerializer
+from ..study.models import Study
 from ..study.serializers import StudyAssessmentSerializer
 from . import models
 
 
 class TaskSerializer(serializers.ModelSerializer):
     owner = HAWCUserSerializer(read_only=True)
     study = StudyAssessmentSerializer(read_only=True)
+    study_id = serializers.PrimaryKeyRelatedField(
```
Comment (rabstejnek) [inline hawc/apps/mgmt/serializers.py:14]: Study is needed for writes; this allows it to be passed in as `study_id`

File: hawc/apps/riskofbias/serializers.py [hunk 5]
```diff
@@ -104,7 +104,7 @@ class Meta:
 
 
 class RiskOfBiasScoreSerializer(serializers.ModelSerializer):
-    overridden_objects = RiskOfBiasScoreOverrideObjectSerializer(many=True)
+    overridden_objects = RiskOfBiasScoreOverrideObjectSerializer(many=True, read_only=True)
```
Comment (rabstejnek) [inline hawc/apps/riskofbias/serializers.py:107]: The override serializer is not really setup for writes, so we just ignore it on writes.

File: hawc/apps/assessment/serializers.py (hunk 1 above)
Comment (shapiromatron) [reply hawc/apps/assessment/serializers.py:67]: I changed to make required; we'll always need to have a parent

File: hawc/apps/lit/serializers.py (hunk 2 above)
Comment (shapiromatron) [reply hawc/apps/lit/serializers.py]: changed to return an empty list instead of null `tree[0].get("children", [])`

**Review by shapiromatron — APPROVED** [review]
Approved; if your ok with my minor change, feel free to merge.

**Pull request merged**